frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1236,7 +1236,9 @@ impl<T: Config> Pallet<T> {
 		}
 
 		// After election finalization, clear OCW solution storage.
-		if <frame_system::Pallet<T>>::events()
+		//
+		// We can read the events here because offchain worker doesn't affect PoV.
+		if <frame_system::Pallet<T>>::read_events_i_know_what_i_am_doing()
```

Comment:
writing a small note here to make it a new issue: This can actually be removed. While it is an offchain read with no PoV consequence or time limit, we can achieve the same by storing the previous phase in some offchain storage, and then checking the new phase here. Whenever `previous_phase != Off && new_phase == Off` we call `unsigned::kill_ocw_solution::<T>();`.